include/bout/coordinates.hxx

```diff
@@ -235,6 +235,9 @@ private:
   /// Set the parallel (y) transform from the options file.
   /// Used in the constructor to create the transform object.
   void setParallelTransform(Options* options);
+
+  void checkCovariant();
+  void checkContravariant();
```

Comment:
Please can you just put a docstring on these? I know it's sort of obvious from the name, but checking _what_ would be good to know

Reply:
```suggestion
  // check that covariant tensors are positive (if expected) and finite (always)
  void checkCovariant();
  // check that contravariant tensors are positive (if expected) and finite (always)
  void checkContravariant();
```